diff --git a/init.rc b/init.rc
index 9fcd53a..a12075f 100644
--- a/init.rc
+++ b/init.rc
@@ -1,9 +1,9 @@
-import /init.b2g.rc
 on early-init
     # Set init and its forked children's oom_adj.
     write /proc/1/oom_adj -16
 
     start ueventd
+    import /init.b2g.rc
 
 # create mountpoints
     mkdir /mnt 0775 root system
@@ -97,6 +97,9 @@ on fs
 # mount mtd partitions
     # Mount /system rw first to give the filesystem a chance to save a checkpoint
     mount yaffs2 mtd@system /system
+    chmod 0755 /system/b2g/b2g
+    chmod 0755 /system/b2g/plugin-container
+    chmod 0755 /system/b2g/updater
     mount yaffs2 mtd@system /system ro remount
     mount yaffs2 mtd@userdata /data nosuid nodev
     mount yaffs2 mtd@cache /cache nosuid nodev
